e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -385,17 +387,22 @@ def _get_climo_dataset(self, season: str) -> xr.Dataset:
         filepath = self._get_climo_filepath(season)
         ds = self._open_climo_dataset(filepath)
 
-        if self.var in ds.variables:
-            pass
-        elif self.var in self.derived_vars_map:
+        if self.var in self.derived_vars_map and (self.var,) in list(
+            self.derived_vars_map[self.var].keys()
+        ):
             ds = self._get_dataset_with_derived_climo_var(ds)
+        elif self.var in ds.variables:
+            pass
```

Comment:
```suggestion
        if self.var in self.derived_vars_map:
            ds = self._get_dataset_with_derived_climo_var(ds)
        elif self.var in ds.variables:
            pass
```

I think we just need to switch up the order of the conditionals to what it was in the old [Dataset._get_climo_var()](https://github.com/E3SM-Project/e3sm_diags/blob/10b16f8a52d5dc606c9060de7b517c01ca4d804b/e3sm_diags/driver/utils/dataset.py#L346-L372)

Any derived variable with a unit conversion formula should run that function.

Reply:
This change might also fix issue #1 from your [comment above](https://github.com/E3SM-Project/e3sm_diags/pull/752#issuecomment-1927819504):

> 1. Variables not generated and saved for `NETCF`, `RESTOM` and `FLDSC`, and all simulator related variables: e.g. `CLDTOT_TAU1.3_ISCCP`. error messsage:
>        `OSError: Variable 'NETCF' was not in the file '/global/cfs/cdirs/e3sm/e3sm_diags/postprocessed_e3sm_v2_data_for_e3sm_diags/20210528.v2rc3e.piControl.ne30pg2_EC30to60E2r2.chrysalis/climatology/rgr/20210528.v2rc3e.piControl.ne30pg2_EC30to60E2r2.chrysalis_ANN_005101_006012_climo.nc', nor was it defined in the derived variables dictionary.`

Reply:
In my PR #774, this update resolves the large relative diffs I was getting for OMEGA and Q variables which require unit conversion for some cases.

Reply:
@tomvothecoder thanks for looking into this issue. with your suggested change, it resolved the first of my 3 remaining regression problems. Though a new CI/CD problem occur, I think it is with the logic of derived variable changed back (shifted order of conditionals)...